tests/MiddleMailServiceTests.cs

```diff
@@ -107,6 +141,27 @@ public async Task ProcessingContinuesAfterGeneralProcessingException() {
 			}
 		}
 
+		[Fact]
+		public async Task RateLimitAllowsCorrectNumberThrough() {
+			for (int i = 0; i < 3; i++) {
+				var emailMessage = FakerFactory.EmailMessageFaker.Generate();
+				await rateLimitedCallback(emailMessage);
+				rateLimitedProcessorMock.Verify(m => m.ProcessAsync(It.IsAny<EmailMessage>()), Times.Exactly(i + 1));
+			}
+
+			// If you want to test that it takes the correct amount of time to delay,
+			// uncomment each commented line,
+			// and change the number of verified sent emails to 4.
+			// DateTime before = DateTime.Now;
+			var notSentMessage = FakerFactory.EmailMessageFaker.Generate();
+			await Task.WhenAny(rateLimitedCallback(notSentMessage), Task.Delay(TimeSpan.FromSeconds(5)));
+			Assert.True(TimeSpan.FromSeconds(5) < MiddleMailService.RateLimitWindow);
```

Comment:
I can add a comment here. The idea is that the test should fail if someone changes the rate limit window. Otherwise, someone could change the rate limit window, then might be confused why the test is sending the email when it shouldn't do that.